src/handlers.rs

```diff
@@ -0,0 +1,166 @@
+use std::io::{Read, Write};
+use std::process::{Command, Stdio};
+use log::{debug, error, trace, warn};
+use teloxide::dispatching::dialogue::InMemStorage;
+use teloxide::utils::command::BotCommands;
+use teloxide::prelude::*;
+use teloxide::RequestError;
+use teloxide::types::{InputFile, ParseMode};
+use tg_graphviz::{Graph, number_of_lines, GraphSyntaxError};
+
+type MyDialogue = Dialogue<State, InMemStorage<State>>;
+
+/// Represents the different states of the bot dialogue with the user
+#[derive(Clone, Default)]
+pub enum State {
+    #[default]
+    Start,
+    ReceiveImage { msg_text: String },
+}
+
+/// Handles the initial state, prompting the user for graph type (directed / undirected)
+pub async fn start(
+    bot: Bot,
+    dialogue: MyDialogue,
+    msg: Message)
+    -> Result<(), RequestError> {
+    if msg.text().is_none() {
+        warn!("User sent NOT MessageCommon, exit..");
+        bot.send_message(msg.chat.id, "It seems that you sent something suspicious to me...\n\
+        Please, don't do that🥺").await?;
+        return Ok(());
+    }
+    trace!("This is message in start state: {}", msg.text().unwrap());
+    bot.send_message(msg.chat.id, "Is your graph directed?(Y\\n)").await?;
+
+    // Passes to the next state
+    dialogue.update(State::ReceiveImage { msg_text: msg.text().unwrap().to_string() }).await.unwrap();
+    Ok(())
+}
+
+/// Handles the state where the bot receives the graph representation
+pub async fn receive_image(
+    bot: Bot,
+    dialogue: MyDialogue,
+    msg_text: String,
+    msg: Message)
+    -> Result<(), RequestError> {
+    if msg.text().is_none() {
+        warn!("User sent NOT MessageCommon, exit..");
+        bot.send_message(msg.chat.id, "It seems that you sent something suspicious to me...\n\
+        Please, don't do that🥺").await?;
+        return Ok(());
+    }
+
+    let direction_choice = msg.text().unwrap().trim();
+    trace!("This is message in receive state: {}", direction_choice);
+
+    // Parses the user's input to determine if the graph is directed
+    let directed: Option<bool> = match direction_choice.to_lowercase().as_str() {
+        "y" => Some(true),
+        "n" => Some(false),
+        _ => None
```

Comment:
Can't you just return here and use `directed: bool`?

Reply:
I agree, this would be better
fixed